source/base/Shmoose.cpp

```diff
@@ -226,13 +233,13 @@ QString Shmoose::getCurrentChatPartner()
     return persistence_->getCurrentChatPartner();
 }
 
-void Shmoose::sendMessage(QString const &toJid, QString const &message, QString const &type)
+void Shmoose::sendMessage(QString toJid, QString message, QString type, QString msgId)
 {
     bool isGroup = rosterController_->isGroup(toJid);
-    messageHandler_->sendMessage(toJid, message, type, isGroup);
+    messageHandler_->sendMessage(toJid, message, type, isGroup, msgId);
 }
 
-void Shmoose::sendMessage(QString const &message, QString const &type)
+void Shmoose::sendMessage(QString message, QString type)
```

Comment:
why not const anymore?

Reply:
Yes I will restore const. I was wondering if passing references is thread safe. Moreover I saw that sometimes QString const& is used and sometimes const QString&.